File: bindings/cpp/src/status.cpp
#include "cppchecks/status.h"
#include "cppchecks/core.h"

namespace CPPCHECKS_NAMESPACE
{
    bool Status::is_pending() const
    {
        return cchecks_status_is_pending((CChecksStatus *)&_value);
    }

    bool Status::has_passed() const
    {
        return cchecks_status_has_passed((CChecksStatus *)&_value);
    }

    bool Status::has_failed() const
    {
        return cchecks_status_has_failed((CChecksStatus *)&_value);
    }

    CChecksStatus Status::c_status() const
    {
        switch (_value)
        {
        case Status::Value::Pending:
            return CChecksStatusPending;
        case Status::Value::Skipped:
            return CChecksStatusSkipped;
        case Status::Value::Passed:
            return CChecksStatusPassed;
        case Status::Value::Warning:
            return CChecksStatusWarning;
        case Status::Value::Failed:
            return CChecksStatusFailed;
        case Status::Value::SystemError:
            return CChecksStatusSystemError;
        }
    }

} // namespace CPPCHECKS_NAMESPACE

File: bindings/python/fuzz/status.py
# ruff: noqa: D103,D100,S101

from __future__ import annotations

import atheris
import hypothesis
import openchecks
from hypothesis import strategies


@hypothesis.given(
    status=strategies.sampled_from(
        [
            openchecks.Status.Pending,
            openchecks.Status.Skipped,
            openchecks.Status.Passed,
            openchecks.Status.Warning,
            openchecks.Status.Failed,
            openchecks.Status.SystemError,
        ]
    ),
)
@atheris.instrument_func
def fuzz(
    status: openchecks.Status,
) -> None:
    assert status.is_pending() == (status == openchecks.Status.Pending)
    assert status.has_passed() == (
        status in [openchecks.Status.Passed, openchecks.Status.Warning]
    )
    assert status.has_failed() == (
        status in [openchecks.Status.Failed, openchecks.Status.SystemError]
    )


if __name__ == "__main__":
    import sys

    atheris.Setup(sys.argv, atheris.instrument_func(fuzz.hypothesis.fuzz_one_input))
    atheris.Fuzz()
